### PR TITLE
[SPARK-28118][CORE] Add `spark.eventLog.compression.codec` configuration

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

Event logs are different from the other data in terms of the lifetime. It would be great to have a new configuration for Spark event log compression like `spark.eventLog.compression.codec` .
This PR adds this new configuration as an optional configuration. So, if `spark.eventLog.compression.codec` is not given, `spark.io.compression.codec` will be used.

## How was this patch tested?

Pass the Jenkins with the newly added test case.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @dbtsai

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 1]
```diff
@@ -98,6 +98,14 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val EVENT_LOG_COMPRESSION_CODEC =
+    ConfigBuilder("spark.eventLog.compression.codec")
+      .doc("The codec used to compress event log. By default, Spark provides four codecs: " +
+        "lz4, lzf, snappy, and zstd. You can also use fully qualified class names to specify " +
+        "the codec. If this is not given, spark.io.compression.codec will be used.")
+      .stringConf
+      .createOptional
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: @dongjoon-hyun can we use fallbackConf instead?

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 2]
```diff
@@ -1157,7 +1165,7 @@ package object config {
 
   private[spark] val IO_COMPRESSION_CODEC =
     ConfigBuilder("spark.io.compression.codec")
-      .doc("The codec used to compress internal data such as RDD partitions, event log, " +
+      .doc("The codec used to compress internal data such as RDD partitions, " +
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: and .. seems like the doc is still valid if `spark.eventLog.compression.codec` is not given.

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by dongjoon-hyun** [comment]
Thank you for review, @HyukjinKwon . I updated the PR according to your comments.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala]: Yep.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala]: Yep. It's recovered.

**Comment by dbtsai** [comment]
LGTM. Thanks.

**Comment by dongjoon-hyun** [comment]
Thank you for review and merge, @dbtsai ! Thank you, @HyukjinKwon , too.

**Pull request closed without merging**